homeassistant/components/sensor/rainbird.py

```diff
@@ -0,0 +1,81 @@
+"""
+Support for Rain Bird Irrigation system LNK WiFi Module.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/sensor.rainbird/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.rainbird import DATA_RAINBIRD
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import CONF_MONITORED_CONDITIONS
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.helpers.entity import Entity
+
+DEPENDENCIES = ['rainbird']
+
+_LOGGER = logging.getLogger(__name__)
+
+# sensor_type [ description, unit, icon ]
+SENSOR_TYPES = {
+    'rainsensor': ['Rainsensor', None, 'water']
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_MONITORED_CONDITIONS, default=list(SENSOR_TYPES)):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up a sensor for a raincloud device."""
+    controller = hass.data[DATA_RAINBIRD]
+
+    sensors = []
+    for sensor_type in config.get(CONF_MONITORED_CONDITIONS):
+        if sensor_type == 'rainsensor':
+            sensors.append(
+                RainBirdSensor(controller, sensor_type))
+
+    add_devices(sensors, True)
+
+
+class RainBirdSensor(Entity):
+    """A sensor implementation for rain bird device."""
+
+    def __init__(self, controller, sensor_type):
+        """Initialize rain bird sensor."""
+        self._sensor_type = sensor_type
+        self._controller = controller
+        self._name = SENSOR_TYPES[self._sensor_type][0]
+        self._icon = 'mdi:{}'.format(SENSOR_TYPES[self._sensor_type][2])
```

Comment:
Let's just keep `mdi:` as part of the configuration dict string. It's confusing to break it up.

Reply:
Changed